File: food-donation-frontend/src/services/notificationService.js
import apiClient from './apiClient';

const notificationService = {
  async getMyNotifications(params = {}) {
    const { data } = await apiClient.get('/notifications', { params });
    return data.data;
  },
  async getUnreadCount() {
    const { data } = await apiClient.get('/notifications/unread-count');
    return data.data.unreadCount;
  },
  async markAsRead(id) {
    const { data } = await apiClient.put(`/notifications/${id}/read`);
    return data;
  },
  async markAllAsRead() {
    const { data } = await apiClient.put('/notifications/read-all');
    return data;
  },
};

export default notificationService;

File: food-donation-backend/src/main/java/com/fooddonation/service/NotificationService.java
package com.fooddonation.service;

import org.springframework.data.domain.Pageable;

import com.fooddonation.dto.response.NotificationResponse;
import com.fooddonation.dto.response.PageResponse;
import com.fooddonation.entity.User;
import com.fooddonation.enums.NotificationType;

public interface NotificationService {

	void sendNotification(User user, String title, String message, NotificationType type, Long referenceId,
			String referenceType);

	PageResponse<NotificationResponse> getMyNotifications(String email, Pageable pageable);

	long getUnreadCount(String email);

	void markAsRead(Long notificationId, String email);

	void markAllAsRead(String email);

}
